docs/DeveloperGuide.md

```diff
@@ -133,6 +133,31 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### List Student Feature
+
+#### Implementation
+The list student feature displays a list of existing students in the TutorBuddy application.
+
+This feature is facilitated by `ListStudentCommand` which extends `Command`.
+The method `ListStudentCommand#execute` updates the filtered student list by calling the method
+`Model#updateFilteredStudentList` exposed in the `Model` interface.
+
+Given below is an example of how the list student 
+1. The user executes the list student command with the input `list_student`.
+2. `LogicManager` executes the input and parses the command using `AddressBookParser`.
+3. `AddressBookParser` identifies the correct command and creates a new `ListStudentCommand`.
+4. `AddressBookParser` returns the new `ListStudentCommand` to  `LogicManager`.
```

Comment:
I think there is additional space on to  `LogicManager`, but small problem ah.

Reply:
Thanks for spotting, will update.